lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -418,63 +418,71 @@ def add_exemption(post, options = {})
 
-        stored_credential = options[:stored_credential]
-        post[:payment_method_options][:card][:mit_exemption] = {}
+        card_options = post[:payment_method_options][:card]
+        card_options[:mit_exemption] = {}
 
         # Stripe PI accepts network_transaction_id and ds_transaction_id via mit field under card.
         # The network_transaction_id can be sent in nested under stored credentials OR as its own field (add_ntid handles when it is sent in on its own)
         # If it is sent is as its own field AND under stored credentials, the value sent under its own field is what will send.
-        post[:payment_method_options][:card][:mit_exemption][:ds_transaction_id] = stored_credential[:ds_transaction_id] if stored_credential[:ds_transaction_id]
-        post[:payment_method_options][:card][:mit_exemption][:network_transaction_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
+        card_options[:mit_exemption][:ds_transaction_id] = stored_credential[:ds_transaction_id] if stored_credential[:ds_transaction_id]
+        card_options[:mit_exemption][:network_transaction_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
+
+        add_stored_credential_transaction_type(post, options)
       end
 
       def add_stored_credential_transaction_type(post, options = {})
+        return unless options[:stored_credential_transaction_type]
+
         stored_credential = options[:stored_credential]
         # Do not add anything unless these are present.
         return unless stored_credential[:reason_type] && stored_credential[:initiator]
 
         # Not compatible with off_session parameter.
         options.delete(:off_session)
-        if stored_credential[:initial_transaction]
-          # Initial transactions must by CIT
-          return unless stored_credential[:initiator] == 'cardholder'
 
-          initial_transaction_stored_credential(post, stored_credential[:reason_type])
-        else
-          # Subsequent transaction
-          subsequent_transaction_stored_credential(post, stored_credential[:initiator], stored_credential[:reason_type])
-        end
+        stored_credential_type = if stored_credential[:initial_transaction]
+                                   return unless stored_credential[:initiator] == 'cardholder'
+
+                                   initial_transaction_stored_credential(post, stored_credential)
+                                 else
+                                   subsequent_transaction_stored_credential(post, stored_credential)
+                                 end
+
+        card_options = post[:payment_method_options][:card]
+        card_options[:stored_credential_transaction_type] = stored_credential_type
+        card_options[:mit_exemption].delete(:network_transaction_id) if stored_credential_type == 'setup_on_session' || options[:setup_future_usage] == 'off_session'
```

Comment:
Would this cover the scenario where we do not send the NTID if `stored_credential_transaction_type` is not passed in AND an NTID is passed in AND `setup_future_usage == off_session`?

Reply:
I thought you said that we wanted to send NTID even if `stored_credential_transaciton_type` is not passed, https://github.com/activemerchant/active_merchant/pull/4828#discussion_r1260162091? At least I interpreted at `stored_credential_transaciton_type: false` being the same thing as it not being passed. 

Let me see if I'm understanding correctly we DO NOT send NTID if 
1. `setup_future_usage == off_session`
2.  `stored_credential_transaction_type ==  setup_on_session`
3.  options doesn't have [:stored_credential_transaction_type]

Reply:
Ahhh I see the confusion, it's scenarios 1 and 2 in that list.

If `stored_credential_transaction_type` doesn't exist in options then we should not add `stored_credential_transaction_type` to the `post` hash. 

For context, the reason we want this change is so that we can send our standard stored credential fields down every time. Right now we only do it on MITs because there's no guard on scenario 1 in AM. There will be a FF gating `stored_credential_transaction_type` and when that is turned on, we'll not pass `setup_future_usage` and instead rely on our stored credential mapping to pick the correct `stored_credential_transaction_type`

Reply:
Okay I think now we're on the same page. We don't add `stored_credential_transaction_type` to post if it doesn't exist in options https://github.com/activemerchant/active_merchant/pull/4828/files#diff-19492101b546175aa3ca6a16ec7d0a610d0adae03cbaf130eff54092dec8e0fbR440. But I'll move the `options[:setup_future_usage] == 'off_session'` from here to the other method and add a test for it. Thanks for all the help!